Detour/Include/DetourCommon.h

```diff
@@ -482,6 +483,15 @@ inline void dtSwapEndian(float* v)
 void dtRandomPointInConvexPoly(const float* pts, const int npts, float* areas,
 							   const float s, const float t, float* out);
 
+template<typename TypeToRetrieveAs, typename BufferType>
+TypeToRetrieveAs* getThenAdvanceBufferPointer(BufferType*& buffer, const size_t distanceToAdvance)
```

Comment:
Couldn't it be `unsigned char*&`? `buffer` is always passed as `unsigned char *` as far as I can see.